src/include/platform/internal/DeviceControlServer.h

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Defines the Device Layer DeviceControlServer object.
+ */
+
+#pragma once
+
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+class DeviceControlServer final
+{
+public:
+    // ===== Members for internal use by other Device Layer components.
+
+    CHIP_ERROR ArmFailSafe(uint16_t expiryLengthSeconds);
+    CHIP_ERROR DisarmFailSafe();
+    CHIP_ERROR CommissioningComplete();
+    CHIP_ERROR SetRegulatoryConfig(uint8_t location, const char * countryCode, uint64_t breadcrumb);
+
+private:
+    // ===== Members for internal use by the following friends.
```

Comment:
do we need these friend declarations? I would prefer to remove any `friend` bits as much as possible to avoid strong coupling.

nit: replace `(void)` with `()`

Reply:
You could have a static member function and then you do not need a friend declaration:

```cpp
static DeviceControlServer &Instance() {
  return sInstance;
}
```

Reply:
Yes, this is another pattern, to use static function to get the instance, we need to call the static member function with class such as DeviceControlServer::DeviceControlSvr().